--- origsrc/virtuoso-opensource-6.1.4/binsrc/driver/Makefile.am	2010-07-09 10:57:25.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/binsrc/driver/Makefile.am	2011-11-24 01:53:00.733199900 -0600
@@ -93,7 +93,7 @@ libvirtodbc_u_la_SOURCES = \
 #  Ansi/Fiber based client
 #
 virtodbc_la_SOURCES = virt_vers.c
-virtodbc_la_LDFLAGS = -module $(VIRTODBC_LDFLAGS) -export-symbols virt.exp
+virtodbc_la_LDFLAGS = -module $(VIRTODBC_LDFLAGS) -export-symbols virt.exp -no-undefined
 virtodbc_la_LIBADD = \
 	libvirtodbc.la \
 	$(VIRTODBC_FIBER_LIBS)
@@ -113,7 +113,7 @@ virtodbc_r_la_LIBADD = \
 #  Unicode/Fiber based client
 #
 virtodbcu_la_SOURCES =  virt_vers_u.c
-virtodbcu_la_LDFLAGS = -module $(VIRTODBC_LDFLAGS) -export-symbols virt_u.exp
+virtodbcu_la_LDFLAGS = -module $(VIRTODBC_LDFLAGS) -export-symbols virt_u.exp -no-undefined
 virtodbcu_la_LIBADD = \
 	libvirtodbc.la \
 	libvirtodbc_u.la \
--- origsrc/virtuoso-opensource-6.1.4/binsrc/hosting/perl/Makefile.am	2007-09-23 12:34:31.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/binsrc/hosting/perl/Makefile.am	2011-11-06 21:58:51.148991600 -0600
@@ -34,7 +34,7 @@ hosting_perl_la_CFLAGS += -I$(top_srcdir
 hosting_perl_la_CFLAGS += -I$(top_srcdir)/libsrc/Wi 
 hosting_perl_la_CFLAGS += -I$(top_srcdir)/libsrc/plugin 
 hosting_perl_la_CFLAGS += $(PERL_CFLAGS)
-hosting_perl_la_LDFLAGS = -module -avoid-version -export-dynamic 
+hosting_perl_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined
 hosting_perl_la_LDFLAGS += $(PERL_LDFLAGS)
 
 virt_handler.c: virt_handler.pm pl_to_c.awk
--- origsrc/virtuoso-opensource-6.1.4/binsrc/hosting/perl/hosting_perl.c	2007-05-04 07:45:19.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/binsrc/hosting/perl/hosting_perl.c	2011-11-24 02:05:07.215898400 -0600
@@ -166,7 +166,7 @@ hosting_perl_version = {
       HOSTING_TITLE,			/*!< Title of unit, filled by unit */
       DBMS_SRV_GEN_MAJOR DBMS_SRV_GEN_MINOR,/*!< Version number, filled by unit */
       "OpenLink Software",			/*!< Plugin's developer, filled by unit */
-      "Perl " PERL_XS_APIVERSION " hosting plugin",			/*!< Any additional info, filled by unit */
+      "Perl hosting plugin",			/*!< Any additional info, filled by unit */
       NULL,					/*!< Error message, filled by unit loader */
       NULL,					/*!< Name of file with unit's code, filled by unit loader */
       hosting_perl_connect,		/*!< Pointer to connection function, cannot be NULL */
--- origsrc/virtuoso-opensource-6.1.4/binsrc/hosting/php/Makefile.am	2010-02-03 13:17:57.000000000 -0600
+++ src/virtuoso-opensource-6.1.4/binsrc/hosting/php/Makefile.am	2011-11-06 21:58:56.952201700 -0600
@@ -36,7 +36,7 @@ hosting_php_la_CFLAGS  += -I$(top_srcdir
 hosting_php_la_CFLAGS  += -I$(top_srcdir)/libsrc/odbcsdk/include 
 hosting_php_la_CFLAGS  +=  $(php5_CFLAGS)
 
-hosting_php_la_LDFLAGS = -module -avoid-version -export-dynamic
+hosting_php_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined
 hosting_php_la_LDFLAGS += -export-symbols hosting_php.exp
 
 
--- origsrc/virtuoso-opensource-6.1.4/binsrc/hosting/python/Makefile.am	2010-04-14 08:55:08.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/binsrc/hosting/python/Makefile.am	2011-11-06 21:59:01.507409700 -0600
@@ -40,7 +40,7 @@ hosting_python_la_CFLAGS += -I$(top_srcd
 hosting_python_la_CFLAGS += -I$(top_srcdir)/libsrc/langfunc 
 hosting_python_la_CFLAGS += $(PYTHON_CFLAGS)
 
-hosting_python_la_LDFLAGS = -module -avoid-version -export-dynamic 
+hosting_python_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined
 hosting_python_la_LDFLAGS += $(PYTHON_LDFLAGS)
 
 virt_handler.c: virt_handler.py py_to_c.awk
--- origsrc/virtuoso-opensource-6.1.4/binsrc/hosting/ruby/Makefile.am	2007-09-23 12:34:31.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/binsrc/hosting/ruby/Makefile.am	2011-11-06 21:58:30.338555300 -0600
@@ -37,7 +37,7 @@ hosting_ruby_la_CFLAGS  += -I$(top_srcdi
 hosting_ruby_la_CFLAGS  += -I$(top_srcdir)/libsrc/langfunc 
 hosting_ruby_la_CFLAGS  +=  $(RUBY_CFLAGS)
 
-hosting_ruby_la_LDFLAGS = -module -avoid-version -export-dynamic 
+hosting_ruby_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined
 hosting_ruby_la_LDFLAGS += $(RUBY_LDFLAGS)
 
 
--- origsrc/virtuoso-opensource-6.1.4/libsrc/plugin/Makefile.am	2009-10-16 11:57:47.000000000 -0500
+++ src/virtuoso-opensource-6.1.4/libsrc/plugin/Makefile.am	2011-11-06 22:01:02.768020900 -0600
@@ -37,7 +37,7 @@ base_plugin_cflags  += -I$(top_srcdir)/l
 base_plugin_cflags  += -I$(top_srcdir)/libsrc/Wi
 base_plugin_cflags  += -I$(top_srcdir)/libsrc/odbcsdk/include
 
-base_plugin_ldflags = -module -avoid-version
+base_plugin_ldflags = -module -avoid-version -no-undefined
 
 
 ##########################################
